ddtrace/tracer/spancontext_test.go

```diff
@@ -855,3 +856,42 @@ func TestTraceIDEmpty(t *testing.T) {
 	tid[15] = 5
 	assert.False(t, tid.Empty())
 }
+
+func TestSpanIDHexEncoded(t *testing.T) {
+	sid := spanIDHexEncoded(5, 16)
+	assert.Equal(t, fmt.Sprintf("%016x", 5), sid)
+
+	sid = spanIDHexEncoded(5, 32)
+	assert.Equal(t, fmt.Sprintf("%032x", 5), sid)
+
+	sid = spanIDHexEncoded(math.MaxInt64, 68)
+	assert.Equal(t, fmt.Sprintf("%068x", math.MaxInt64), sid)
+
+	sid = spanIDHexEncoded(math.MaxInt64, 128)
+	assert.Equal(t, fmt.Sprintf("%0128x", math.MaxInt64), sid)
+}
+
+func BenchmarkSpanIDHexEncoded(b *testing.B) {
+	for n := 0; n < b.N; n++ {
+		_ = spanIDHexEncoded(32, 16)
+	}
+}
+
+func BenchmarkSpanIDSprintf(b *testing.B) {
+	for n := 0; n < b.N; n++ {
+		_ = fmt.Sprintf("%016x", 32)
+	}
+}
+
+func FuzzSpanIDHexEncoded(f *testing.F) {
+	f.Fuzz(func(t *testing.T, p int, v uint64) {
+		expected := fmt.Sprintf(
+			fmt.Sprintf("%%0%dx", p),
+			v,
+		)
+		actual := spanIDHexEncoded(v, p)
+		if actual != expected {
+			t.Fatalf("expected %s, got %s", expected, actual)
```

Comment:
This failed for me with `p = -99`, `v = 0`:

```
--- FAIL: FuzzSpanIDHexEncoded (0.26s)
    --- FAIL: FuzzSpanIDHexEncoded (0.00s)
        spancontext_test.go:894: expected 0                                                                                                  , got 0
```

Actual `fmt` does right-padding when the width is negative. I don't think that's needed for this implementation, so perhaps just guard this test body with `if p < 0 { return }`?

Reply:
Negative paddings seem to be accepted by `fmt.Sprintf`, and handled as if padding is zero, so it makes sense to handle them in the test body or in the function. I'd rather do the second to keep the same behaviour in the function.

Reply:
If I highlight that output with my mouse, I see spaces at the end. Try putting something after it, e.g. https://go.dev/play/p/vQbOAmAtCMT: `fmt.Printf("%-10dhello", 42)` outputs `42        hello`